packages/forms/src/Components/TableBuilder.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Filament\Forms\Components;
+
+use Closure;
+use Filament\Support\Concerns\HasExtraAlpineAttributes;
+
+class TableBuilder extends Field
+{
+    use HasExtraAlpineAttributes;
+    use Concerns\CanLimitItemsLength;
+
+    protected string $view = 'forms::components.table-builder';
+
+    protected string | Closure $addColumnButtonLabel;
+
+    protected string | Closure $addRowButtonLabel;
+
+    protected int | Closure $defaultRows = 1;
+
+    protected int | Closure $defaultColumns = 1;
+
+    protected bool | Closure $shouldDisableAddingRows = false;
```

Comment:
What about renaming these to be consistent with the rest of the codebase?
```suggestion
    protected bool | Closure $isAddingRowsDisabled = false;
```

Reply:
$canAddRows ?